test/e2e/set/scyllacluster/scyllacluster_pv.go

```diff
@@ -5,35 +5,254 @@ package scyllacluster
+						o.Expect(err).NotTo(o.HaveOccurred())
+
+						if len(realPvc.Spec.VolumeName) != 0 {
+							return false, nil
+						}
+
+						pv, err := f.KubeAdminClient().CoreV1().PersistentVolumes().Get(ctx, pvc.Spec.VolumeName, metav1.GetOptions{})
+						o.Expect(err).NotTo(o.HaveOccurred())
+
+						pvClone := &corev1.PersistentVolume{
+							ObjectMeta: metav1.ObjectMeta{
+								Name:   fmt.Sprintf("clone-%s", pv.Name),
+								Labels: f.CommonLabels(),
+							},
+							Spec:   *pv.Spec.DeepCopy(),
+							Status: *pv.Status.DeepCopy(),
+						}
+						pvClone.Labels[cloneLabelKey] = f.Namespace()
+						pvClone.Spec.StorageClassName = testStorageClassName
+						pvClone.Spec.NodeAffinity = nil
+						pvClone.Spec.ClaimRef = nil
+						if pvClone.Spec.PersistentVolumeSource.Local != nil {
+							pvClone.Spec.PersistentVolumeSource.HostPath = &corev1.HostPathVolumeSource{
+								Path: pvClone.Spec.PersistentVolumeSource.Local.Path,
+							}
+							pvClone.Spec.PersistentVolumeSource.Local = nil
+						}
+
+						framework.Infof("Creating clone PV for %q", pv.Name)
+						_, err = f.KubeAdminClient().CoreV1().PersistentVolumes().Create(ctx, pvClone, metav1.CreateOptions{})
+						o.Expect(err).NotTo(o.HaveOccurred())
+
+						realPvcCopy := realPvc.DeepCopy()
+						realPvcCopy.Spec.VolumeName = pvClone.Name
+
+						framework.Infof("Binding ScyllaCluster PVC %q with clone PV %q", realPvc.Name, pvClone.Name)
+						_, err = f.KubeClient().CoreV1().PersistentVolumeClaims(realPvc.Namespace).Update(ctx, realPvcCopy, metav1.UpdateOptions{})
+						o.Expect(err).NotTo(o.HaveOccurred())
+					}
+				case watch.Deleted:
+					// If original PVC is deleted, delete the clone too.
+					pvc := e.Object.(*corev1.PersistentVolumeClaim)
+					if strings.HasPrefix(pvc.Name, "clone-") {
+						return false, nil
+					}
+
+					clonePVCName := fmt.Sprintf("clone-%s", pvc.Name)
+					clonePVC, err := f.KubeClient().CoreV1().PersistentVolumeClaims(pvc.Namespace).Get(ctx, clonePVCName, metav1.GetOptions{})
+
+					framework.Infof("Deleting clone PVC %q", clonePVCName)
+					err = f.KubeClient().CoreV1().PersistentVolumeClaims(f.Namespace()).Delete(ctx, clonePVCName, metav1.DeleteOptions{})
+					o.Expect(err).NotTo(o.HaveOccurred())
+
+					consumerPodName := fmt.Sprintf("consumer-%s", clonePVCName)
+					framework.Infof("Deleting Pod consumer of clone PVC %q", consumerPodName)
+
+					err = f.KubeClient().CoreV1().Pods(f.Namespace()).Delete(ctx, consumerPodName, metav1.DeleteOptions{})
+					o.Expect(err).NotTo(o.HaveOccurred())
+
+					err = framework.WaitForObjectDeletion(ctx, f.DynamicAdminClient(), corev1.SchemeGroupVersion.WithResource("persistentvolumeclaims"), f.Namespace(), clonePVC.Name, &clonePVC.UID)
+				}
+				return false, nil
+			})
+			if err != nil {
+				if err == wait.ErrWaitTimeout && ctx.Err() == context.Canceled {
+					return
+				}
+				o.Expect(err).NotTo(o.HaveOccurred())
+			}
```

Comment:
```suggestion
			if err == wait.ErrWaitTimeout && ctx.Err() == context.Canceled {
				return
			}
			o.Expect(err).NotTo(o.HaveOccurred())
```

Reply:
looking at your new diff, I'd like to clarify that my main point was that the `err != nil` is not needed https://github.com/scylladb/scylla-operator/compare/57b442b8cbd451a23248cfa17458776d6ec3017f..d4c47854e620282d2bee9c368afb1c0607e1a85a#diff-c9a98932b51802ef0c02382487f46ec72ddf800486f802b17ebbf4dd32873867R249

Reply:
good point, got rid of it